### PR TITLE
Add Validation::isScalar()

### PR DESCRIPTION
**Pull request by chinpei215**

### CODE REVIEW

File: tests/TestCase/Validation/ValidationTest.php [hunk 1]
```diff
@@ -2879,6 +2879,18 @@ public function testIsArray()
...
 
     /**
+     * Test isScalar
+     *
+     * @return void
+     */
+    public function testIsScalar()
+    {
+        $this->assertTrue(Validation::isScalar(1));
+        $this->assertTrue(Validation::isScalar(''));
```
Comment (inoas) [inline tests/TestCase/Validation/ValidationTest.php]: + `$this->assertTrue(Validation::isScalar(true));`

Comment (chinpei215) [inline tests/TestCase/Validation/ValidationTest.php]: I am lazy 😛

File: src/Validation/Validator.php [hunk 2]
```diff
@@ -1619,6 +1619,24 @@ public function isArray($field, $message = null, $when = null)
     }
 
     /**
+     * Add a validation rule to ensure that a field contains a scalar.
```
Comment (dereuromark) [inline src/Validation/Validator.php]: Should we add a small note that it means? That it would not be an object or array?

File: src/Validation/Validation.php [hunk 3]
```diff
@@ -1424,6 +1424,17 @@ public static function isArray($value)
...
 
     /**
+     * Check that the input value is a scalar.
+     *
+     * @param mixed $value The value to check
+     * @return bool
+     */
+    public static function isScalar($value)
+    {
+        return is_scalar($value);
```
Comment (dereuromark) [inline src/Validation/Validation.php]: Please add a test for NULL, afaik NULL needs special handling as it is not "scalar" according to this check, but in our scope it should be OK to pass IMO.

Comment (chinpei215) [inline src/Validation/Validation.php]: I am find to add tests for nulls. But in my opinion, this method should not pass null, because null is not a scalar, and if I remember correctly, null values are not passed to validation methods and handled specially in Validator class.

File: src/Validation/Validator.php (hunk 2 above)
Comment (chinpei215) [inline src/Validation/Validator.php]: I think scalar is scalar in PHP, and there is no ambiguity.

File: src/Validation/Validation.php (hunk 3 above)
Comment (chinpei215) [inline src/Validation/Validation.php]: To be clear, I am talking about a case of `allowEmpty(true)`.

Comment (inoas) [inline src/Validation/Validation.php]: Imho it should NOT behave different than `is_scalar()`; But we can have `isScalarOrNull()` or `isScalarOrNullish()` (latter accepts '' and null)?

Comment (chinpei215) [inline src/Validation/Validation.php]: We don't have `isIntegerOrNull()`, and `isInteger()` doesn't pass null values. So I don't think we need to have `isScalarOrNull()`.

File: src/Validation/Validator.php (hunk 2 above)
Comment (dereuromark) [inline src/Validation/Validator.php]: Validation rules are still high level and for a user just seeing that rule it might still be valuable not having to check the php internal manuals maybe :)

File: tests/TestCase/Validation/ValidationTest.php (hunk 1 above)
Comment (inoas) [inline tests/TestCase/Validation/ValidationTest.php]: but after the discussion below: `$this->assertFalse(Validation::isScalar(null));`... maybe ;)

File: src/Validation/Validator.php (hunk 2 above)
Comment (chinpei215) [inline src/Validation/Validator.php]: OK, I can add a note.

File: tests/TestCase/Validation/ValidationTest.php (hunk 1 above)
Comment (chinpei215) [inline tests/TestCase/Validation/ValidationTest.php]: Probably I would add the test too.

**Comment by chinpei215** [comment]
Oh I didn't notice that the proxy method for `Validation::isInteger()` is `Validator::integer()`. Should the proxy method for `Validation::isScalar()` be `Validator::scalar()`?

**Comment by chinpei215** [comment]
What do you think about `Validation::isScalar()`? Most methods in Validation class doesn't have `is` prefix. Would `Validation::scalar()` and `Validator::scalar()` be better?

**Comment by dereuromark** [comment]
wasnt the issue that integer is a reserved keyword?

**Comment by chinpei215** [comment]
Though `array` is a reserved keyword, but `integer` and `scalar` are not. So we can name `Validator::integer()`. And in PHP7, we can use `array` for a method name without using `__call()` hack. So I think it is unecessary to care about reserved keywords if it is not reserved at this time.

**Comment by saeideng** [comment]
`isScalar` is good naming

**Comment by markstory** [comment]
:+1: for `isScalar()` as a method name for the validator.

**Comment by chinpei215** [comment]
OK. Then I would rename `Validator::isScalar()` only.
